### ISSUE TITLE
jackson and jaxb polymorphism

### ISSUE THREAD

**Issue by Pumuckels** [stoicflame/enunciate#961]
Hello,

I use Jaxb for XML-mapping and Jackson for Json-mapping. The data structure contains dynamic polymorphism. 
More specific, I have a tree element that represents either a node or leaf. A node itself can contain other nodes or leafes.

```java
@XmlRootElement
@JacksonXmlRootElement( localName = "request" )
public class TreeRequest
{ 
   @DocumentationExample( type = @TypeHint( Node.class ) )
   @XmlElement
   @JsonTypeInfo( use = NAME, include = WRAPPER_OBJECT )
   @JsonSubTypes( {
	@JsonSubTypes.Type( value = Leaf.class, name = "leaf" ),
	@JsonSubTypes.Type( value = Node.class, name = "node" )
   } ) 
   private TreeElem root;
}

public class Node implements TreeElem
{
   
   @DocumentationExample( type = @TypeHint( Leaf.class ) )
   @xmlElement
   @JsonTypeInfo( use = NAME, include = WRAPPER_OBJECT )
   @JsonSubTypes( {
	@JsonSubTypes.Type( value = Leaf.class, name = "leaf" ),
	@JsonSubTypes.Type( value = Node.class, name = "node" )
   } )
   private List<TreeElem> elements;

}

public interface TreeElem
{
}
```
The generated json documentation contains the expected hierarchy. So the request contains one node and the node contains a leaf. 
So the expected request would look like;
```json
{
   "root": {
      "node": {
	  "nodeName": "test"		
	}	
    }
}
```
AND
```json
`{
	"root": {
		"node": {
			"nodeName": "test",
			"elements" : [ {
				"leaf" : {
					"entityId" : 12345
				}
			}, {
				"node" : {
					"entityId" : 12345
				}
			} ]
		}
    }
}
```

But the generate xml documentation is missing the name of the node and leaf.  
```xml
<root>
	<nodeName>test</nodeName>
</root>
```
AND 

```xml
<root>
	<nodeName>test</nodeName>
	...
	<elements>
		<entityId>12345</entityId>
	</elements>
</root>
```

Is it possible to only define the @JsontypeInfo & @JsonSubTypes in the interface TreeElem, but not have to define it for the fields holding that interface?

``` java
@DocumentationExample( type = @TypeHint( Leaf.class ) )
   @JsonTypeInfo( use = NAME, include = WRAPPER_OBJECT )
   @JsonSubTypes( {
	@JsonSubTypes.Type( value = Leaf.class, name = "leaf" ),
	@JsonSubTypes.Type( value = Node.class, name = "node" )
   } )
public interface TreeElem
{
}

public class Node implements TreeElem
{
   @DocumentationExample( type = @TypeHint( Leaf.class ) )
   @XmlExample
   private List<TreeElem> elements;
}

```

Is it possible to get the same result without changing anything about the data structure for Jaxb?
 
Basically its the same problem as described in https://github.com/stoicflame/enunciate/issues/716

Thanks

**Comment by stoicflame** [comment]
> Is it possible to only define the @JsontypeInfo & @JsonSubTypes in the interface TreeElem, but not have to define it for the fields holding that interface?

Not that I know of. Have you asked the Jackson community?

> Is it possible to get the same result without changing anything about the data structure for Jaxb?

Again, not that I know of. Have you asked the JAXB community?

**Comment by Pumuckels** [comment]
Thank you for the response. The creation of the json document works, unfortunately encunciate can not implement the polymorphism for the xml: 

Example: 
current documentation output:

<root>
	<nodeName>test</nodeName>
	...
	<elements>
		<entityId>12345</entityId>
	</elements>
</root>

expected documentation output:

<root>
	<node>
		<nodeName>test<nodeName>
		...
		<elements>
			<leaf>
				<entityId>12345</entityId>
			</leaf>
			<node>
			    <entityId>12345</entityId>
			</node>
		</elementS>
	<node>
<root>

**Comment by stoicflame** [comment]
I'm still not sure what you're asking. Are you asking how to use JAXB annotations to get the desired XML output?

If that's the question, a more fitting place to ask that would be over at the JAXB project. But I can offer my personal knowledge if it's worth anything. My best guess as to what you want would be:

```java
public class Node implements TreeElem
{
   @DocumentationExample( type = @TypeHint( Leaf.class ) )
   @XmlExample
   @XmlElementWrapper(name = "elements")
   @XmlElements({
       @XmlElement(name="leaf", type=Leaf.class),
       @XmlElement(name="node", type=Node.class)
   })
   private List<TreeElem> elements;
}
```
